pkg/controller/core.oam.dev/v1alpha2/application/apply.go

```diff
@@ -197,6 +225,10 @@ func (h *appHandler) createOrUpdateComponent(ctx context.Context, comp *v1alpha2
 	// object is persisted as Raw data after going through api server
 	updatedComp := comp.DeepCopy()
 	updatedComp.Spec.Workload.Object = nil
+	if updatedComp.Spec.Helm != nil {
+		updatedComp.Spec.Helm.Release.Object = nil
+		updatedComp.Spec.Helm.Repository.Object = nil
```

Comment:
why set to nil?

Reply:
as comments in line 224, the object in raw extension will be persistent as raw data after processed by API server.